src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -208,15 +208,16 @@ class TaskBuilder(
   }
 
   protected def computeContainerInfo(hostPorts: Seq[Option[Int]]): Option[ContainerInfo] = {
-    if (runSpec.container.isEmpty && runSpec.ipAddress.isEmpty) None
-    else {
+    if (runSpec.container.isEmpty && runSpec.ipAddress.isEmpty) {
+      None
+    } else {
       val builder = ContainerInfo.newBuilder
 
       // Fill in Docker container details if necessary
       runSpec.container.foreach { c =>
-        val portMappings = c.docker.map { d =>
-          d.portMappings.map { pms =>
-            pms.zip(hostPorts).collect {
+        val containerWithPortMappings = c match {
+          case docker: Container.Docker => docker.copy(pms =
```

Comment:
In general: Yes, please! In this particular case, it's hard to change. Here, a new Container is created (by copying) that may have different port mappings. Because `Container` has been changed to be a trait, the copy has to be created from the actual implementation. Right now, only `Container.Docker` supports port mappings, hence only this case is considered here. If/When other containers implementations get support for port mappings, these cases have to be implemented here as well.